app/src/main/java/pm/gnosis/heimdall/data/repositories/TokenRepository.kt

```diff
@@ -4,11 +4,12 @@ import io.reactivex.Completable
 import io.reactivex.Flowable
 import io.reactivex.Observable
 import pm.gnosis.heimdall.data.repositories.model.ERC20Token
+import java.math.BigInteger
 
 interface TokenRepository {
     fun observeTokens(): Flowable<List<ERC20Token>>
-    fun observeTokenInfo(token: ERC20Token): Observable<ERC20Token>
-    fun addToken(address: String, name: String = ""): Completable
-    fun removeToken(address: String): Completable
+    fun loadTokenInfo(contractAddress: BigInteger): Observable<ERC20Token>
```

Comment:
maybe we should change our retrofit calls to Singles, else this one is weird with our pattern "load" for single vs "observe" for abservable